applications/plugins/CGALPlugin/CGALPlugin_test/MeshFromImageWithFeatures_test.cpp

```diff
@@ -0,0 +1,222 @@
+    //typedef component::container::MechanicalObject<DataTypes> MechanicalObject;
+    
+    
+    /*typedef typename MechanicalObject::VecCoord  VecCoord;
+    typedef typename MechanicalObject::Coord  Coord;
+    typedef typename MechanicalObject::VecDeriv  VecDeriv;
+    typedef typename MechanicalObject::Deriv  Deriv;*/
+    //typedef typename DataTypes::Real  Real;
+
+    bool test() //double /*epsilon*/, const std::string &/*integrationScheme */)
+    {
+        cgal::MeshGenerationFromImage< defaulttype::Vec3dTypes, defaulttype::ImageUC >* meshGenerator;
+
+        meshGenerator = dynamic_cast<cgal::MeshGenerationFromImage< defaulttype::Vec3dTypes, defaulttype::ImageUC >*>(root->getObject("generator"));
+
+        helper::ReadAccessor< Data<sofa::defaulttype::Vec3dTypes::VecCoord> > features = meshGenerator->d_features;
+        helper::ReadAccessor< Data<sofa::defaulttype::Vec3dTypes::VecCoord> > nodes = meshGenerator->f_newX0;
+
+        double tolerance = 1e-10;
+        size_t nfts = features.size();
+        bool foundFeatures[nfts];
+        std::cout << "#features to be tested: " << nfts << std::endl;
+        for (size_t fi = 0; fi < nfts; fi++) {
+            foundFeatures[fi] = false;
+            const defaulttype::Vec3dTypes::Coord& ft = features[fi];
+            for (size_t i = 0; i < nodes.size() && !foundFeatures[fi]; i++) {
+                const defaulttype::Vec3dTypes::Coord & np = nodes[i];
+
+                double dist = 0.0;
+                for (size_t d = 0; d < 3; d++)
+                    dist += helper::SQR(np[d] - ft[d]);
+                dist = sqrt(dist);
+
+                if (dist < tolerance) {
+                    //std::cout << "Feature: " << features[fi] << std::endl;
+                    //std::cout << "Node: [" << i << "]: " << np << std::endl;
+                    foundFeatures[fi] = true;
+                }
+            }
+        }
+
+        for (size_t fi = 0; fi < nfts; fi++) {
+            if (!foundFeatures[fi])
+                return false;
+        }
+
+        return true;
+
+	}
+
+    void testDataFields()
+    {
+        std::cout << "Calling test fields!" << std::endl;
+        /*EXPECT_EQ(grabber->d_fileName.getName(), "videoFile");
+        EXPECT_EQ(grabber->d_seekFrame.getName(), "seek");
+        EXPECT_EQ(grabber->d_camIdx.getName(), "cam_index");
+        EXPECT_EQ(grabber->d_paused.getName(), "paused");
+        EXPECT_EQ(grabber->d_stopped.getName(), "stopped");
+
+        EXPECT_EQ(grabber->d_videoMode.getName(), "video_mode");
+        EXPECT_EQ(grabber->d_fullFrame.getName(), "fullFrame");
+        EXPECT_EQ(grabber->d_frame1.getName(), "img1");
+        EXPECT_EQ(grabber->d_frame2.getName(), "img2");
+        EXPECT_EQ(grabber->d_fps.getName(), "fps");
+        EXPECT_EQ(grabber->d_dimensions.getName(), "dimensions");*/
+    }
+
+    void SetUp()
+    {
+        /// a scene defined using raw-strings (C++11)
```

Comment:
+1 for the raw-string...this look super great way to write scene in tests. Thank for that tip.